src/ProposalTypesConfigurator.sol

```diff
@@ -164,12 +164,59 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
         if (!_proposalTypesExists[proposalTypeId]) revert InvalidProposalType();
 
         if (!scopeExists[proposalTypeId][txTypeHash]) revert InvalidScope();
-        Scope[] memory validScopes = scopes[proposalTypeId];
+        Scope memory validScope = scopes[proposalTypeId][txTypeHash];
+        return validScope.encodedLimits;
+    }
+
+    function getParameter(bytes calldata limit, uint256 startIdx, uint256 endIdx)
+        public
+        pure
+        returns (bytes memory parameter)
+    {
+        return limit[startIdx:endIdx + 1];
+    }
+
+    function validateProposedTx(bytes calldata proposedTx, uint8 proposalTypeId, bytes32 txTypeHash)
+        public
+        view
+        returns (bool valid)
+    {
+        Scope memory validScope = scopes[proposalTypeId][txTypeHash];
+        bytes memory scopeLimit = validScope.encodedLimits;
+        bytes4 selector = bytes4(scopeLimit);
+        if (selector != bytes4(proposedTx[:4])) revert Invalid4ByteSelector();
+
+        uint256 startIdx = 4;
+        uint256 endIdx = 0;
+        for (uint8 i = 0; i < validScope.parameters.length; i++) {
+            if (i == 0) {
+                endIdx = startIdx + validScope.parameters[i].length;
+            } else {
+                endIdx = endIdx + validScope.parameters[i].length;
+            }
+
+            bytes32 param = bytes32(proposedTx[startIdx:endIdx]);
```

Comment:
we should state in natspec comments where in this contract that this version of proposal types only supports bytes32-type parameters, and not `bytes`